src/Controls/StreetControls.js

```diff
@@ -77,8 +113,74 @@ class StreetControls extends FirstPersonControls {
         this.previousPosition = newPreviousPosition;
     }
 
+    pick(event, pickGround = () => {}, pickObject = () => {}, target = new THREE.Vector3()) {
+        // get real distance to ground, with a specific method to pick on the elevation layer
+        this.view.getPickingPositionFromDepth(this.view.eventToViewCoords(event), target);
+        const distanceToGround = this.camera.position.distanceTo(target);
+
+        const intersectsBuildings = this.view.pickObjectsAt(event, 0, this.buildingsLayer);
+        const intersectsGround = this.view.pickObjectsAt(event, 0, this.view.tileLayer);
+
+        // to detect pick on building, compare first picked building distance to ground distance
+        if (intersectsBuildings.length && intersectsBuildings[0].distance < distanceToGround) { // pick buildings
+            this.updateSurface(intersectsBuildings[0].point, intersectsBuildings[0].face.normal, this.surfaceWall, this.surfaceGround);
```

Comment:
There are certain functions that do not need to be exposed (`updateSurface, onMouseUp`...)

Reply:
Yes I moved some of them